Services/DeviceClient.cs

```diff
@@ -103,13 +105,30 @@ public class DeviceClient : IDeviceClient
 
         public async Task SendMessageAsync(string message, DeviceModel.DeviceModelMessageSchema schema)
         {
-            var eventMessage = new Message(Encoding.UTF8.GetBytes(message));
+            var eventMessage = default(Message);
+
+            switch (schema.Format)
+            {
+                case (DeviceModel.DeviceModelMessageSchemaFormat.JSON):
+                    eventMessage = new Message(Encoding.UTF8.GetBytes(message));
+                    eventMessage.ContentType = "application/json";
+                    eventMessage.ContentEncoding = "utf-8";
+                    break;
+                case (DeviceModel.DeviceModelMessageSchemaFormat.PROTO):
+                    switch (schema.Name)
+                    {
+                        case ("GPSPacket;v1"):
```

Comment:
This approach won't scale and would require listing every possible class available.  

As per other comment, use the new `ClassName` device model property to specify which class to use, searching the class in the assembly with reflection.  We should consider the cost of reflection, but that can be done separately to keep this PR small.

1. get class name T from `MessageTemplate`
2. check if T exists, e.g. `typeof(IServicesConfig).GetTypeInfo().Assembly.GetTypes().Where(...)`
3. if T doesn't exists, exception
4. if T exists use JsonConvert.DeserializeObject(message, T)